File: #2090.k-radius-subarray-averages.cpp
class Solution {
public:
    vector<int> getAverages(vector<int>& nums, int k) {

        int size = nums.size();
        vector<long long> sums(size, 0);
        for(int i=0; i<size; i++) {
            sums[i] = nums[i] + ((i == 0) ? 0 : sums[i - 1]); 
        }

        vector<int> results(size, -1);
        for(int i = k; i <= size - k - 1; i++) {
            results[i] = (sums[i + k] - ((i - k - 1) < 0 ? 0 : sums[i - k - 1])) / (2 * k + 1);
        }
        return results;
    }
};

File: #2090.k-radius-subarray-averages.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number[]}
 */
var getAverages = function(nums, k) {
    
    let rad = k * 2 + 1;
    let arr = Array(nums.length).fill(-1);
    let start = 0;
    let sum = 0;
    for (let i = 0; i < nums.length; i++) {
        sum += nums[i];
        if(i >= rad - 1) {
            arr[Math.floor((i + start) / 2)] =  Math.floor(sum / rad);
            sum -= nums[start];
            start++;
        }
    }
    return arr;
};
